File: Java/src/com/dbc/algorithm/Search/FindMinimumInRotatedSortedArray.java
package com.dbc.algorithm.Search;

public class FindMinimumInRotatedSortedArray {
    public int findMin(int[] nums) {
        int length = nums.length;
        int left = 0, right = length - 1, res = 50001;
        while (left <= right) {
            int mid = (left + right) / 2;
            if (nums[mid] <= nums[right]) {
                right = mid - 1;
            } else if (nums[mid] >= nums[left]) {
                left = mid + 1;
            }
            res = Math.min(res, nums[mid]);
        }
        return res;
    }
}

File: core/tmp/Python/search/find_minimum_in_rotated_sorted_array.py
from typing import *

def findMin(nums: List[int]) -> int:
    length = len(nums)
    left, right = 0, length - 1
    res = 50001
    while left <= right:
        mid = (left + right) // 2
        if nums[mid] <= nums[right]:
            right = mid - 1
        elif nums[mid] >= nums[left]:
            left = mid + 1
        res = min(res, nums[mid])
    
    return res

if __name__ == "__main__":
    print(findMin([11,13,15,17]))